src/Components/Components/src/ComponentBase.cs

```diff
@@ -157,6 +157,38 @@ protected Task Invoke(Action workItem)
         protected Task InvokeAsync(Func<Task> workItem)
             => _renderHandle.InvokeAsync(workItem);
 
+        /// <summary>
+        /// Dispatches a state change notification to the provided component.
+        /// </summary>
+        /// <param name="component">The component instance.</param>
+        /// <param name="invoker">The <see cref="EventHandlerInvoker"/></param>
+        /// <param name="args">The event args.</param>
+        /// <returns>
+        /// A <see cref="Task" /> that asynchronously completes once event processing has completed.
+        /// </returns>
+        public static Task DispatchStateChangeAsync(object component, EventHandlerInvoker invoker, UIEventArgs args)
```

Comment:
Could the public versions of `DispatchStateChangeAsync` take an `IComponent component`, instead of `object component`? Seems like that would make the usage more explicit. I don't think there's any scenario where we expect people to dispatch to something that's not a component.

It's fine if there's also an internal overload that takes `object component` so we don't have to put casts and checks into all the `SetValueHandler` calls.

Reply:
Or even `IHandleStateChange` instead of `IComponent`.

Reply:
Also the overload that takes `(object component, EventHandlerInvoker invoker, UIEventArgs args)` seems only to be used in one place, and that place could call `invoker.Invoke(args)` itself then use the `(object component, Task task)` overload. So maybe we only need the second overload.

Reply:
Sure. I cribbed this from your original PR - but I think the semantics have changed. You originally had this typed as object or `IComponent` so that callers could at least attempt to dispatch a change manually with minimal ceremony.

Another consideration is whether we want to allow AppState objects to implement `IHandleStateChange`.

Reply:
I think I'd prefer to leave this internal or let it accept object. There are **a lot** of call sites of this, and it doesn't seem useful to expand the boilerplate into all of those locations.

Reply:
This is gone now, so not super relevant.